Allow the modification of config using LethalConfig

MasterAli2 opened the pull request:
Updates done to the config at runtime will now change the resolution and other stuff to be changed.

On lethal config it will still say that it requires restart but it doesn't, that's because I don't know how to add a reference without possibly breaking stuff.

If there is any bugs regarding this it should not affect players that didn't modify the config with lethal config.

An issue that I seem to inconsistently get is if I change the resolution to something huge like 5K or 6K and set it back to default it wont change but that should be fixed with addition of `defaultScreenTexHeight` and `defaultTerminalTexHighResHeight `

I'm not sure if this requires test for monitors of different aspect.

This should close #2

Someone else should probably also check it works cuz I don't trust myself.

MasterAli2 committed:
feat: update resolution when config is modified at runtime

Resolution and other config will now be updated when modified at runtime (LethalConfig)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/LethalCompanyModding/LCUltrawide?pullRequest=7) <br/>All committers have signed the CLA.

MasterAli2 committed:
fix: lower cyclomatic complexity

lowered cyclomatic complexity to pass deepsource check

no downsides to this other than an update is required if said variables are modified in a future game update

RobynLlama commented:
I don't know how this mod works on what the expected behavior is so I can't test it but maybe @Lunxara can build this version and let me know if it breaks anything, at least?

MasterAli2 commented:
I forgot to add something to ignore configResW when it's modified with LethalConfig.


I will probably do that later today.

MasterAli2 committed:
Merge branch 'LethalCompanyModding:main' into main

MasterAli2 committed:
fix: make sure configResW is 0 when modified with LethalConfig

RobynLlama commented:
This all certainly sounds good but I need somebody to test it and tell me it works before I merge it. Again, I don't even use this mod, I just keep it from breaking between versions so I don't even know what it does, ahaha

MasterAli2 commented:
I understand, I don't mind waiting

RobynLlama commented:
Request merged, Lunxara confirmed using Lethal Config seems to work and I added the features to the changelog and pushed a new version to the thunderstore
